dsc/item_submission.py

```diff
@@ -0,0 +1,68 @@
+import json
+import logging
+from dataclasses import dataclass
+from typing import Any
+
+from dsc.utilities.aws.s3 import S3Client
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ItemSubmission:
```

Comment:
Do you think `ItemSubmission` would benefit from some kind of `validate()` method?  I'm unsure offhand what that might look for, but I'm noticing now that we call `generate_and_upload_dspace_metadata()` in the `for` loop of the workflow and it might be helpful to check things look good before we fire it off.

Reply:
I did have a `valid_dspace_metadata` method in `wiley-deposits` but I may want to rework it since that had specific field names to validate while these workflows may be less constrained. But yes, there should be some validation!

Reply:
Side note: If we find that `dataclass` module is the right fit for this module, I propose calling the proposed `validate()` method in a `__post_init__` method!